django_loci/static/django-loci/js/loci.js

```diff
@@ -462,22 +464,25 @@ django.jQuery(function ($) {
                 if (!$addressInput.val()) {
                     $addressInput.val(data.address);
                 } else {
-                    message = gettext('Do you wish to change this address?');
+                    message = gettext('The location on the map was changed, would you ' +
+                                      'like to update the address to');
+                    message += ' "' + data.address + '"?';
                     if (confirm(message)) {
                         $addressInput.val(data.address);
                     }
                 }
             })
             .fail(function () {
-                message = gettext('Could not find address related to the location.');
+                message = gettext('Could not find any address related to this location.');
                 alert(message);
             });
     }
 
-    // drag marker event
-    function updateAddrOnDrag() {
+    // triggers update of the address when the location on the map is changed
+    function updateAddressOnMapChange() {
         var marker = getMarker();
-        marker.on('dragend', function () {
+        if (!marker) { return; }
+        getMap().on('draw:edited', function (e) {
```

Comment:
@NoumbissiValere @oltarzewskik :eyes: :point_up: https://leaflet.github.io/Leaflet.draw/docs/leaflet-draw-latest.html#l-draw-event

Reply:
Nice, `edited` is cool. it fires up the callback only the save option is clicked which is perfect. thank you @nemesisdesign 
:+1: